### DISCUSSION TITLE
HLS -> RTSP convert stream

### DISCUSSION THREAD

**Discussion by mjimeneznet** [bluenviron/mediamtx#829] [Questions] [answered]
Hello, I have a Netatmo Presence Camera which streams in HLS format with an URL like `http://192.168.1.1/af983498sdf98sadf/live/files/high/index.m3u8` 

When I open the stream with `ffmpeg` this is the stream info:

```[hls @ 0x55732e4a0280] Opening 'http://192.168.1.1/af983498sdf98sadf/live/files/high/live0001519454.ts' for reading
[hls @ 0x55732e4a0280] Opening 'http://192.168.1.1/af983498sdf98sadf/live/files/high/live0001519455.ts' for reading
Input #0, hls, from 'http://192.168.1.1/af983498sdf98sadf/live/files/high/index.m3u8':
  Duration: N/A, start: 80152.751644, bitrate: N/A
  Program 0 
    Metadata:
      variant_bitrate : 0
  Stream #0:0: Video: h264 (Main) ([27][0][0][0] / 0x001B), yuv420p, 1920x1080, 24 fps, 24 tbr, 90k tbn, 180k tbc
    Metadata:
      variant_bitrate : 0
  Stream #0:1: Audio: aac (LC) ([15][0][0][0] / 0x000F), 16000 Hz, mono, fltp
    Metadata:
      variant_bitrate : 0
```

What I would like to do is convert this HLS Stream into a RTSP Stream, so I can configure that RTSP stream into my Dahua NVR for recording that camera. My NVR doesn't support HLS as input, but it suports RTSP / ONVIF input.

Can you help me to convert from HLS to RTSP using rtsp-simple-server?

Thanks!

**Comment by aler9** [comment, marked as answer]
This is pretty straightforward, you can use this configuration:

```yml
paths:
  mycamera:
    source: http://192.168.1.1/af983498sdf98sadf/live/files/high/index.m3u8
```

After starting the server, the stream will be available at `rtsp://localhost:8554/mycamera`.

**Comment by tlowcode** [comment]
@mjimeneznet did you manage setup a working stream? My approaches always end after a couple segments read with an error.
[https://github.com/aler9/rtsp-simple-server/discussions/1366](url)

**Reply by mjimeneznet** [reply]
Yes It did. The solution proposed by the author worked for me

[a comment by an automated account is not shown]
